<?xml version="1.0" ?>
<sdf version="1.7">
  <world name="default">
    <scene>
      <ambient>0.4 0.4 0.4 1.0</ambient>
      <background>0.7 0.7 0.7 1.0</background>
      <shadows>true</shadows>
    </scene>
    
    <physics default="0" name="default_physics" type="ode">
      <max_step_size>0.001</max_step_size>
      <real_time_factor>1</real_time_factor>
      <real_time_update_rate>1000</real_time_update_rate>
    </physics>
    
    <plugin filename="libignition-gazebo-physics-system.so" name="ignition::gazebo::systems::Physics"/>
    <plugin filename="libignition-gazebo-scene-broadcaster-system.so" name="ignition::gazebo::systems::SceneBroadcaster"/>
    <plugin filename="libignition-gazebo-user-commands-system.so" name="ignition::gazebo::systems::UserCommands"/>
    <plugin filename="libignition-gazebo-sensors-system.so" name="ignition::gazebo::systems::Sensors">
      <render_engine>ogre2</render_engine>
    </plugin>
    
    <include>
      <uri>https://fuel.gazebosim.org/1.0/OpenRobotics/models/Warehouse</uri>
    </include>
    
    <light name="warehouse_light_0" type="point">
      <pose frame="">0 0 9 0 0 0</pose>
      <diffuse>0.5 0.5 0.5 1</diffuse>
      <specular>0.2 0.2 0.2 1</specular>
      <attenuation>
        <range>80</range>
        <constant>0.3</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <cast_shadows>1</cast_shadows>
      <direction>0.1 0.1 -1</direction>
    </light>
    
    <light name="warehouse_light_1" type="point">
      <pose>5 5 8 0 0 0</pose>
      <diffuse>0.7 0.7 0.7 1</diffuse>
      <specular>0.1 0.1 0.1 1</specular>
      <attenuation>
        <range>20</range>
        <constant>0.5</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
    </light>
    
    <light name="warehouse_light_2" type="point">
      <pose>-5 5 8 0 0 0</pose>
      <diffuse>0.7 0.7 0.7 1</diffuse>
      <specular>0.1 0.1 0.1 1</specular>
      <attenuation>
        <range>20</range>
        <constant>0.5</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
    </light>
    
    <light name="warehouse_light_3" type="point">
      <pose>5 -5 8 0 0 0</pose>
      <diffuse>0.7 0.7 0.7 1</diffuse>
      <specular>0.1 0.1 0.1 1</specular>
      <attenuation>
        <range>20</range>
        <constant>0.5</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
    </light>
    
    <light name="warehouse_light_4" type="point">
      <pose>-5 -5 8 0 0 0</pose>
      <diffuse>0.7 0.7 0.7 1</diffuse>
      <specular>0.1 0.1 0.1 1</specular>
      <attenuation>
        <range>20</range>
        <constant>0.5</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
    </light>   
    
    <model name="realsense_camera_1">
      <static>true</static>
      <pose>0 10 3.5 0 0.3 -1.57</pose>
      <link name="camera_link">
        <visual name="cam_1_visual">
          <geometry>
            <box>
              <size>0.025 0.09 0.025</size>
            </box>
          </geometry>
          <material>
            <ambient>0.2 0.2 0.2 1</ambient>
            <diffuse>0.2 0.2 0.2 1</diffuse>
          </material>
        </visual>
        <collision name="cam_1_collision">
          <geometry>
            <box>
              <size>0.025 0.09 0.025</size>
            </box>
          </geometry>
        </collision>
        <sensor name="color" type="camera">
          <always_on>1</always_on>
          <update_rate>30</update_rate>
          <visualize>true</visualize>
          <camera>
            <horizontal_fov>1.211</horizontal_fov>
            <image>
              <width>640</width>
              <height>480</height>
              <format>R8G8B8</format>
            </image>
            <clip>
              <near>0.1</near>
              <far>100</far>
            </clip>
          </camera>
          <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
            <ros>
              <namespace>camera_1</namespace>
              <argument>image_raw:=color/image_raw</argument>
              <argument>camera_info:=color/camera_info</argument>
            </ros>
            <camera_name>color</camera_name>
            <frame_name>camera_1_color_optical_frame</frame_name>
          </plugin>
        </sensor>
        <sensor name="depth" type="depth">
          <always_on>1</always_on>
          <update_rate>30</update_rate>
          <visualize>true</visualize>
          <camera>
            <horizontal_fov>1.047</horizontal_fov>
            <image>
              <width>640</width>
              <height>480</height>
              <format>R_FLOAT32</format>
            </image>
            <clip>
              <near>0.1</near>
              <far>10</far>
            </clip>
          </camera>
          <plugin name="depth_camera_controller" filename="libgazebo_ros_camera.so">
            <ros>
              <namespace>camera_1</namespace>
              <argument>image_raw:=depth/image_raw</argument>
              <argument>camera_info:=depth/camera_info</argument>
            </ros>
            <camera_name>depth</camera_name>
            <frame_name>camera_1_depth_optical_frame</frame_name>
          </plugin>
        </sensor>
        <sensor name="pointcloud" type="depth">
          <always_on>1</always_on>
          <update_rate>30</update_rate>
          <camera>
            <horizontal_fov>1.047</horizontal_fov>
            <image>
              <width>640</width>
              <height>480</height>
              <format>R_FLOAT32</format>
            </image>
            <clip>
              <near>0.1</near>
              <far>10</far>
            </clip>
          </camera>
          <plugin name="pointcloud_controller" filename="libgazebo_ros_depth_camera.so">
            <ros>
              <namespace>camera_1</namespace>
            </ros>
            <camera_name>pointcloud</camera_name>
            <frame_name>camera_1_depth_optical_frame</frame_name>
            <point_cloud_cutoff>0.1</point_cloud_cutoff>
          </plugin>
        </sensor>
      </link>
    </model>
    
    <model name="realsense_camera_2">
      <static>true</static>
      <pose>8 0 3.0 0 0.2 3.14</pose>
      <link name="camera_link">
        <visual name="cam_2_visual">
          <geometry>
            <box>
              <size>0.025 0.09 0.025</size>
            </box>
          </geometry>
          <material>
            <ambient>0.2 0.2 0.2 1</ambient>
            <diffuse>0.2 0.2 0.2 1</diffuse>
          </material>
        </visual>
        <collision name="cam_2_collision">
          <geometry>
            <box>
              <size>0.025 0.09 0.025</size>
            </box>
          </geometry>
        </collision>
        <sensor name="color" type="camera">
          <always_on>1</always_on>
          <update_rate>30</update_rate>
          <visualize>true</visualize>
          <camera>
            <horizontal_fov>1.211</horizontal_fov>
            <image>
              <width>640</width>
              <height>480</height>
              <format>R8G8B8</format>
            </image>
            <clip>
              <near>0.1</near>
              <far>100</far>
            </clip>
          </camera>
          <plugin name="camera_controller" filename="libgazebo_ros_camera.so">
            <ros>
              <namespace>camera_2</namespace>
              <argument>image_raw:=color/image_raw</argument>
              <argument>camera_info:=color/camera_info</argument>
            </ros>
            <camera_name>color</camera_name>
            <frame_name>camera_2_color_optical_frame</frame_name>
          </plugin>
          
        </sensor>
        <sensor name="pointcloud" type="depth">
          <always_on>1</always_on>
          <update_rate>30</update_rate>
          <camera>
            <horizontal_fov>1.047</horizontal_fov>
            <image>
              <width>640</width>
              <height>480</height>
              <format>R_FLOAT32</format>
            </image>
            <clip>
              <near>0.1</near>
              <far>10</far>
            </clip>
            
          
          </camera>
          
          
          <plugin name="pointcloud_controller" filename="libgazebo_ros_depth_camera.so">
            
            
            <ros>
              <namespace>camera_2</namespace>
            </ros>
            
            
            <camera_name>pointcloud</camera_name>
            
            
            <frame_name>camera_2_depth_optical_frame</frame_name>
            
            
            <point_cloud_cutoff>0.1</point_cloud_cutoff>
            
          
          </plugin>
          
        
        </sensor>
      </link>
    </model>
    <model name="realsense_camera_3">
      <static>true</static>
      <pose>-6 -6 2.8 0 0.15 0.785</pose>
      <link name="camera_link">
        <visual name="cam_3_visual">
          <geometry>
            <box>
              <size>0.025 0.09 0.025</size>
            </box>
          </geometry>
          
          
          <material>
            <ambient>0.2 0.2 0.2 1</ambient>
            <diffuse>0.2 0.2 0.2 1</diffuse>
          </material>
          
        
        </visual>
        
        
        <collision name="cam_3_collision">
          <geometry>
            <box>
              <size>0.025 0.09 0.025</size>
            </box>
          </geometry>
        </collision>
        

        
        
        <sensor name="pointcloud" type="depth">
          <always_on>1</always_on>
          <update_rate>30</update_rate>
          
          
          <camera>
            
            
            <horizontal_fov>1.047</horizontal_fov>
            
            
            <image>
              <width>640</width>
              <height>480</height>
              <format>R_FLOAT32</format>
            </image>
            
            
            <clip>
              <near>0.1</near>
              <far>10</far>
            </clip>
            
          
          </camera>
          
          
          <plugin name="pointcloud_controller" filename="libgazebo_ros_depth_camera.so">
            
            
            <ros>
              <namespace>camera_3</namespace>
            </ros>
            
            
            <camera_name>pointcloud</camera_name>
            
            
            <frame_name>camera_3_depth_optical_frame</frame_name>
            
            
            <point_cloud_cutoff>0.1</point_cloud_cutoff>
            
          
          </plugin>
          
        
        </sensor>
      </link>
    </model>
  </world>
</sdf>
